### ISSUE TITLE
Option to toggle changelog opening in new tab on update.

### ISSUE THREAD

**Issue by BenMcGarry** [honestbleeps/Reddit-Enhancement-Suite#3397]
This has been asked for a lot as a lot of people just do not care about the changes, I think it should be easy enough to implement. Just a option to turn off changelogs opening in a new tab on update.

**Comment by larsjohnsen** [comment]
Finding some way to show the changelog in `RESAnnouncementAlert` may preferable, as using both seems redundant.

**Comment by honestbleeps** [comment]
There are reasons not to make this an option, we should discuss before doing work.

**Comment by Ajedi32** [comment]
Rather than just randomly opening a new tab (which can be rather jarring, not to mention confusing), it might be better to show some kind of notification when RES updates with an option to view the changelog.

**Comment by honestbleeps** [comment]
let's be fair/honest here: "jarring" is an awfully strong word for a background tab opening up.  it's not like it takes away your focus or anything.

**Comment by Ajedi32** [comment]
It's in the background? For some reason I thought I remembered it being a foreground tab...

**Comment by honestbleeps** [comment]
yep, it's definitely background.

**Comment by l27-0-0-1** [comment]
Opening it in background messes _is_ jarring considering that chrome updates extensions at seemingly random times. I had a situation multiple times when I open some tabs in background from news sites to read later and when I do, I discover some random patchnotes that I couldn't care less about in that stack. Honestly, I see no reason to even show the changelog at all, most of the people probably close it as it opens, and people who do want to read it would find it on their own.

**Comment by trlkly** [comment]
Every other extension I have either doesn't display a notice, or offers a way to shut them off.  Not even the web browsers themselves force you to read the change notes. 

I note that RES already puts out a notification the Reddit userbar when a big new update comes down. People who are interested can click that. Plus, that way they see the notice when they are using Reddit, and not just randomly. 

Plus, when it comes to new features, it seems you are now (brilliantly) putting a gear icon to let us quickly shut them off if we don't like them. That means we don't really need to be told about those features. We'll discover them on our own.

**Comment by honestbleeps** [comment]
There are other reasons why we open the what's new page besides telling people about new features, which I realize some of you just flat out aren't going to like, but they are reality:

- When users have no idea that RES updated, our tech support requests are likely going to be more confused, poorly written and frequent than they already are

- The what's new page is our ONLY source of analytics for knowing how many users we have and on what mix of browsers, etc - in part because the mere mention of throwing analytics into the extension, even with an opt-out, was met with rage (and please, please do not suggest opt-in analytics, that defeats the entire purpose of analytics - it wouldn't be nearly as useful)

- The what's new page is (well, was, we still need to add it back on the new website) our only vehicle for actively pointing people to the fact that they can contribute to RES (via money and/or code).  Yes "people can seek that out if they want to", but the amount of donations RES sees is already extremely paltry as is - and while this is absolutely not a money making venture, being able to provide beer/coffee money to the people busting their ass to make this product better is not something I'm willing to just write off / forget.  We already know anecdotally that the vast majority of RES users rarely/never open up the console, so even that isn't a great spot to draw attention there. We could, in theory, add a different sort of nag, but tying it to "work was done to make RES better" seems sensible in terms of timing.

We're talking about a background tab that doesn't steal focus from what you're currently doing, and it opens somewhere between maybe 3-6 times per **year** here. The information / benefit it provides us, in my opinion, far outweighs the vocal complaints of annoyance from a passionate few people.

The vast majority of the time, I'm all about listening to the users and giving them what they want - but sometimes I feel like people are overblowing really tiny things, and we're going to create a culture of "everyone who yells enough gets what they want" when in reality we should be prioritizing things based on a combination of users needs, depth/breadth of impact, and also our own needs as developers.

**Comment by allthefoxes** [comment]
With @honestbleeps here - I see no reason to add this option. It seems most people complaining about this suffer from the firefox bug anyways, which adding this setting will be redundant.

I think the only good alternative is to make it show in a popup like a np warning.

RES is a very _intimate_ extension. The most out of any I use, the next closest thing I have is my custom ad list in my ad-blocker.

RES however stores all sorts of things and is highly customizable. The way my filters work, the way links are colored, my accounts, my comment colors, more and more and more. Other extensions might not do this often, but I don't need my extension that refreshes all tabs to do that. Its just a button I press. With RES, its all about me. I need to know what has changed. 

Not only that, but as @honestbleeps has said, **many** people forget RES is even a thing. Literally multiple times a day, "Is that an RES feature?" "What is RES?...oh yeah I have that."

etc...

Letting people know we exist, where they can get help, where they can contribute, is important. 

Anything you do will annoy someone. It seems to have annoyed a small handful. Job well done

**Comment by Ajedi32** [comment]
Thanks, that's a pretty good explanation. I realize this is a fairly minor detail, so I won't make too big of a deal out of it if your mind's made up, but I do feel like there's almost certainly better ways of accomplishing those goals than a random "what's new" page opening in the background whenever RES updates. Specifically:

1. *Tech Support**: I feel like a notification appearing within Reddit itself would actually serve this goal *better* than the current solution, as it would be putting the same information in a far more relevant context than a new tab. It'd also give you the flexibility of experimenting with various levels of intrusiveness to help balance user experience with user education. Also, a big banner notice in `/r/resissues` after a release might also help with this.

2. **Analytics**: This seems like a more difficult issue, as I get the feeling there's a lot of history/drama surrounding this that I'm probably unaware of, but isn't the current solution really just a roundabout way of implementing "send usage statistics to RES with no opt-out everytime an update is triggered"? Personally, I've never really understood the desire some people have to avoid sharing anonymous usage statistics with developers, but I recognise that some (though probably a minority, even among Reddit users) are very passionate about that particular issue. If even an opt-out isn't satisfactory, maybe some kind of opt-in/opt-out notification that pops up every once in a while until users make a decision would work? Or maybe a notification that gives users several opportunities to opt-out before collecting anonymous statistics becomes the default?

3. **Contributions**: Wouldn't an in-page notification serve this same purpose? You could probably even make it better by emphasizing the fact that donations and contributions in the form of code are welcome.

Imagine something like this appearing on Reddit after RES updates:

> ### Reddit Enhancement Suite has just updated to v5.0.0!
>
> This is a **major** release with many new features, enhancements, and bug fixes. Here are some of the highlights:
>
> * Microsoft Edge Support (huge thanks BenMcGarry and erikdesjardins)
> * Massive infrastructure overhauls towards faster development, faster startup times (massive thanks erikdesjardins, mc10)
> * IndexedDB storage for Firefox: more stable than simple-storage (thanks erikdesjardins)
> * Chrome storage improvements: increased performance and unlimited capacity (previously 5 MB) (thanks erikdesjardins)
> * Show Images overhaul (huge thanks larsjohnsen)
> * i.redd.it and redditmedia image support
> * A ridiculous amount of hard work from erikdesjardins, mc10, larsjohnsen, githue, matheod, thybag
> * Contributions from many other members in the community
>
> [See the full list of changes](http://redditenhancementsuite.com/2016/09/01/500/) | [View current known issues with this release](http://redditenhancementsuite.com/2016/09/01/500-issues/)
> 
> RES is entirely free - as in beer, as in [open source](https://github.com/honestbleeps/Reddit-Enhancement-Suite), as in everything. If you like our work, a contribution would be greatly appreciated. 
>
> [Donate](http://redditenhancementsuite.com/contribute/) | [Contribute Code](https://github.com/honestbleeps/Reddit-Enhancement-Suite)
>
> [Dismiss](#dismiss)

And like I said, you could vary the size, shape, and UX of that notification to balance user experience with education, user contributions, etc.

**Comment by honestbleeps** [comment]
I can't fathom a scenario where the same people complaining about a background tab that doesnt steal focus from what you're doing are suddenly ok with a growl or other ui popup. They'd likely complain even more.

**Comment by Ajedi32** [comment]
Yeah that's a good point. Actually, now that I think about it, I think my main problem with the background page right now isn't really so much that it's intrusive as it is that it's not immediately obvious what it's for. (It's just opening the changelog, right? I'm not really sure how to trigger the update behavior manually, so I'm just going off of memory right now.)

[The page it opened for 5.0.0](http://redditenhancementsuite.com/2016/09/01/500/) doesn't even mention "Reddit Enhancement Suite" on the page anywhere immediately obvious, nor does it say that RES just updated - it just gives a random list of updates with no context. At the very least, a dedicated page for announcing the update should be created, maybe with wording similar to what I suggested above.

**Comment by trlkly** [comment]
I see a lot of problems with the reasoning here. First off, asking for a
feature that nearly every other addon has is not "blowing things out of
proportion." It's just a request.

Second, you can get those same analytics without opening a page. Just use
an xmlhttprequest. Or, if you need to get past the Cross Site Origin
policy, use JS to load an image but never display it. (It doesn't even have
to actually be an image--the server will still log it, and the request will
just run the onerror function.) Plus Mozilla at least provides at least
that basic level of analytics and I'd be shocked if Google doesn't. So
you'd only need data for other browsers.

Third, the people who don't like the popunder are unlikely to donate, since
they will simply close the ad and not read it. So it probably won't affect
revenue. And there are many other places you could put a Support RES link
to make up any small difference. The dropdown would make sense, as would
the settings console.

Fourth, if it's just a few users, then they are unlikely to affect much
even if you didn't do any of the above stuff I mention. I see no mention
of  actual testing to even see if any of your concerns are actually valid.
You don't actually know if revenue will drop or how many people are willing
to turn the feature off.

Fifth, if you annoy users, then they are less likely to contribute. I have
no other addon that hijacks my browser and forces a popunder, and I use a
lot of addons. A couple have it on by default, but I can and have turned it
off. Popunders are a thing so hated by users that web browsers instituted
built in blocks. Only recently have sites been working around this--but
most of us block them with adblock.

In conclusion, I don't really don't think it's nearly as bad as you have
portrayed it here. There are other ways to get the minimal analytics, and
people who always close the popunder aren't going to donate anyways. Plus
you have no real data to support your claims.

RES is all about removing annoyances. I hope you will allow users to remove
this one.

**Comment by l27-0-0-1** [comment]
Wait a second, so the reason you don't want to make popup disable-able is because you use it as a telemetry tool, but because of semantics you can use it to gather data and also claim there is no telemetry in the extension, while also not giving users option to disable said telemetry? That's some flawed logic right there.

**Comment by BenMcGarry** [comment]
Closed. See "RES Welcome Wagon" within the settings console.
